src/Compiler.js

```diff
@@ -17,6 +17,13 @@ export default function Compiler(data_dir) {
   );
 
   async function compile(code) {
+    let args;
+    try {
+      args = code.split("\n")[0].split("/vala ")[1].split(" ");
```

Comment:
can you move this to its own function (in the same file) and document what it does